File: Python/58-对称的二叉树.py
# -*- coding:utf-8 -*-
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    def isSymmetrical(self, pRoot):
        # write code here
        if not pRoot:
            return True
        
        def helper(a, b):
            if not a or not b:
                return a == b
            return a.val == b.val and helper(a.left, b.right) and helper(a.right, b.left)
        
        return helper(pRoot.left, pRoot.right)

File: C++/58-对称的二叉树.cpp
/*
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) :
            val(x), left(NULL), right(NULL) {
    }
};
*/
class Solution
{
public:
    bool helper(TreeNode *a, TreeNode *b)
    {
        if(!a || !b)
            return a == b;
        return a->val == b->val && helper(a->left, b->right) && helper(a->right, b->left);
    }
    bool isSymmetrical(TreeNode *pRoot)
    {
        if(!pRoot)
            return true;
        return helper(pRoot->left, pRoot->right);
    }
};